src/modules/typenames.ts

```diff
@@ -68,21 +37,24 @@ export function formatTypeNames(query: IQuery) {
   };
 }
 
-export function gankTypeNamesFromResponse(response: object) {
+export function gankTypeNamesFromResponse(response: IExecutionData) {
   const typeNames = [];
   getTypeNameFromField(response, typeNames);
   return typeNames.filter((v, i, a) => a.indexOf(v) === i);
 }
 
-function getTypeNameFromField(obj: object, typenames: string[]) {
+function isObject(obj): obj is IExecutionData {
```

Comment:
This typecheck is not strictly correct as it’s casting to the type containing the typename field, but the actual typename check is outside of it.

Then again if we remove said type and contain it to this file only it’s not as relevant